indico/modules/rb_new/client/js/common/bookings/BookingDetails.jsx

```diff
@@ -294,6 +294,34 @@ class BookingDetails extends React.Component {
         });
     };
 
+    transformToLabel = (type) => {
+        switch (type) {
+            case 'bookings':
+                return {label: Translate.string('Current booking'), style: 'booking', order: 1};
+            case 'cancellations':
+                return {label: Translate.string('Cancelled'), style: 'cancellation', order: 2};
+            case 'rejections':
+                return {label: Translate.string('Rejected'), style: 'rejection', order: 3};
+            case 'other':
+                return {label: Translate.string('Other bookings'), style: 'other', order: 4};
+            default:
+                return undefined;
+        }
+    };
+
+    getLegendLabels = (availability) => {
+        const legendLabels = [];
+        Object.entries(availability).forEach(([type, occurrences]) => {
```

Comment:
An alternative that would be probably faster... (?)
```js
const legendLabels = Object.entries(availability).reduce((accum, [type, occurrences]) => {
    //...
   return accum;
}, {});
```
... and just set `accum[label] = true` if there's a match. The fact that we'd use an object would make the membership check O(1) instead of O(N).
It's not a big deal since the set of labels will always be bound to be small. So, maybe not worth the effort...